clippy_lints/src/methods/mod.rs

```diff
@@ -3142,7 +3143,23 @@ fn is_maybe_uninit_ty_valid(cx: &LateContext<'_, '_>, ty: Ty<'_>) -> bool {
     }
 }
 
-fn lint_suspicious_map(cx: &LateContext<'_, '_>, expr: &hir::Expr<'_>) {
+fn lint_suspicious_map<'tcx>(cx: &LateContext<'_, 'tcx>, expr: &hir::Expr<'tcx>, map_arg: &'tcx hir::Expr<'_>) {
+    match map_arg.kind {
+        hir::ExprKind::Path(_) => {
+            // TODO: How to detect that we have a mutating function path?
```

Comment:
You could try the [`maybe_body_owned_by`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc/hir/map/struct.Map.html#method.maybe_body_owned_by) method to get the body of  the closure and then proceed like below.

Reply:
Maybe you could even write

```rust
if let Some(body_id) = cx.tcx.hir().maybe_body_owned_by(expr.hir_id) {
    // ...
}
```

directly, instead of the match.

Reply:
That indeed works. Thanks.